### DISCUSSION TITLE
Fallback to new_os_window if new_tab can't run

### DISCUSSION THREAD

**Discussion by RobJDavey** [kovidgoyal/kitty#5784] [Q&A] [answered]
I am using the shortcut <kbd>⌘</kbd> <kbd>N</kbd> to open new tabs, however if there is no existing window (but kitty is still the current foreground app) this shortcut does nothing.
Ideally, I'd like it to call `new_os_window` if there is no existing tab, therefore allowing a single shortcut to always result in a new tab.

Is there any syntax for the `map` command to allow this to happen?

**Comment by kovidgoyal** [comment]
map cmd+n launch --type=tab

**Reply by RobJDavey** [reply]
Is there additional configuration that is required for this to work?
When I try this with no window I still get no action happening.
If I run with `--debug-input`, I can see it is ignoring the command because there is no window:
``` shell
-------------- flags changed -----------------
NSEvent: type=FlagsChanged loc=(584.969,653.563) time=144160.7 flags=0x100108 win=0x0 winNum=0 ctxt=0x0 keyCode=55
flagsChanged ignored as no keyWindow present
---------------- key down -------------------
NSEvent: type=KeyDown loc=(584.969,653.563) time=144161.7 flags=0x100108 win=0x0 winNum=0 ctxt=0x0 chars="n" unmodchars="n" repeat=0 keyCode=45
keyDown ignored as no keyWindow present
----------------- key up --------------------
NSEvent: type=KeyUp loc=(584.969,653.563) time=144161.8 flags=0x100108 win=0x0 winNum=0 ctxt=0x0 chars="n" unmodchars="n" repeat=0 keyCode=45
keyUp ignored as no keyWindow present
-------------- flags changed -----------------
NSEvent: type=FlagsChanged loc=(584.969,653.563) time=144162.8 flags=0x100 win=0x0 winNum=0 ctxt=0x0 keyCode=55
flagsChanged ignored as no keyWindow present
```

**Comment by kovidgoyal** [comment, marked as answer]
Ah, right, macOS. You need an entry in the global menubar for this to
work as only those actions are triggered when no window is present. Just
use this
https://github.com/kovidgoyal/kitty/commit/6efb5dc2c20f8beee086def75ab5efd5b85331a5
